src/common/controllers/ads-controller.js

```diff
@@ -339,6 +367,39 @@ class AdsController extends FakeEventTarget implements IAdsController {
     });
   }
 
+  _handleReturnToLive(adBreaks: Array<RunTimeAdBreakObject>) {
+    this._liveEventManager.listenOnce(this._player, AdEventType.AD_ERROR, () => {
+      this._pushNextAdsForLive(adBreaks);
+      this._liveEventManager.removeAll();
+    });
+    this._liveEventManager.listenOnce(this._player, AdEventType.AD_BREAK_END, () => {
+      this._liveEventManager.listenOnce(this._player, Html5EventType.SEEKED, () => {
+        this._pushNextAdsForLive(adBreaks);
+        this._liveEventManager.removeAll();
+      });
+      this._player.seekToLiveEdge();
+    });
+  }
+
+  _pushNextAdsForLive(adBreaks: Array<RunTimeAdBreakObject>) {
+    AdsController._logger.debug('Pushing next ads for live', adBreaks);
+    const liveConfigAdBreaks = [];
+    adBreaks.forEach(adBreak => {
+      const {every, ads} = adBreak;
+      const position = (this._player.isOnLiveEdge() ? this._player.currentTime : adBreak.position) + every;
+      const nextAdBreak = {
+        every,
+        position,
+        ads,
+        played: false,
+        loadedPromise: Promise.resolve()
+      };
+      AdsController._logger.debug('Pushing next ad for live', nextAdBreak);
+      liveConfigAdBreaks.push(nextAdBreak);
+    });
+    this._configAdBreaks = liveConfigAdBreaks;
```

Comment:
same Q like above

Reply:
No, I always keep just one ad in this array (the next one).
Since it's to manipulating the original array during the foreach loop, you must use a new one and then update the reference.